static/app/views/onboarding/targetedOnboarding/setupDocs.tsx

```diff
@@ -178,13 +189,12 @@ function SetupDocs({
       project_id: newProjectId,
     });
     browserHistory.push(`${window.location.pathname}?${searchParams}`);
-    if (clientState) {
```

Comment:
nit: IMO prefer the if statment here

Reply:
I thought the opposite 🤷

Reply:
My rule is, if you're not returning the value don't use `&&`.

There might be a lint rule for this

Reply:
I feel like we use `&&` frequently to handle cases where we have to do to make TS happy for a case that should never happen (if `clientState` is null then we'd redirect the user anyways).

Reply:
It's not a true flow control statement, it's just for type protection

Reply:
I mean, sounds like a true flow control statment, `clientState` could be null, we want flow control to stop that right?

Reply:
@evanpurkhiser It's never gonna happen, we'd never call `setNewProject` if `clientState` is null

Reply:
Maybe I should return an automatic redirect if `clientState` is null

Reply:
sounds like something my be wrong with the control flow of `clientState` farther up in the code then if TS thinks it might still be null 😏

Reply:
@evanpurkhiser I was wrong about `clientState` being null, it's null while the component is hydrating. This is why we need to build out better support for aync hooks. What we need to do is delay loading until the client state is loaded. If we loaded this via `AsyncComponent`, it would be handled automatically.

Reply:
so it's typed correctly then 😏